public_asset/lib/data/network/public_asset_api.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:public_asset/domain/exceptions/public_asset_exceptions.dart';
+import 'package:uuid/uuid.dart';
+
+class PublicAssetApi {
+  const PublicAssetApi(this._httpClient, this._uuid);
+
+  final HttpClient _httpClient;
+  final Uuid _uuid;
+
+  void _checkForSupportedCapability(Session session) {
+    if (!session.capabilities.containsKey(CapabilityIdentifier.jmapPublicAsset)) {
+      throw const UnsupportedCapabilityException();
+    }
+  }
+
+  MapEntry<Id, PatchObject> _toPatchObjectMapEntry(PublicAsset publicAsset) {
+    final patchObject = PatchObject({
+      PatchObject.identityIdsProperty: const PublicAssetIdentitiesConverter()
+        .toJson(publicAsset.identityIds!),
+    });
+
+    return MapEntry(publicAsset.id!, patchObject);
+  }
+
+  Future<List<PublicAsset>> getPublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<Id> publicAssetIds}
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = GetPublicAssetMethod(accountId);
+    method.addIds(publicAssetIds.toSet());
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAssets = response.parse<GetPublicAssetResponse>(
+        invocation.methodCallId,
+        GetPublicAssetResponse.deserialize
+    )?.list ?? [];
+
+    return publicAssets;
+  }
+
+  Future<PublicAsset> createPublicAsset(
+    Session session,
+    AccountId accountId,
+    {
+      required Id blobId,
+      required IdentityId identityId
+    }
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    final generateCreateId = Id(_uuid.v1());
+    method.addCreate(
+      generateCreateId,
+      PublicAsset(blobId: blobId, identityIds: {identityId: true}));
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAsset = response.parse<SetPublicAssetResponse>(
+        invocation.methodCallId,
+        SetPublicAssetResponse.deserialize
+    )?.created?[generateCreateId];
+
+    if (publicAsset == null) {
+      throw const CannotCreatePublicAssetException();
+    }
+
+    return publicAsset;
+  }
+
+  Future<void> destroyPublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<Id> publicAssetIds}
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    method.addDestroy(publicAssetIds.toSet());
+    await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
```

Comment:
How do we know it will succeed without checking the response?

Reply:
We are not concerning if the request is success or not, identity is still updated regardless of the result.

Reply:
from my point of view, we should respect the logic of api. And move the logic of handling success or failure back to use-case. Ok for the case of idenity, but we dont know the other cases with public asset in the future.

Reply:
Done